### PR TITLE
feature(eip): Add transform EIP

### PR DESCRIPTION
**Pull request by igarashitm**
Adding transform EIP (#130)

Expecting this to work once the following is merged
https://github.com/KaotoIO/camel-component-metadata/pull/3

### CODE REVIEW

**Comment by igarashitm** [comment]
hmm kaoto-backend doesn't have camel-component-metadata as a dependency... where does the step-catalog come from then?

File: kamelet-support/src/main/java/io/kaoto/backend/api/service/step/parser/kamelet/KameletStepParserService.java [hunk 1]
```diff
@@ -296,6 +299,22 @@ private Step processDefinedStep(final SetHeaderFlowStep step) {
         return res;
     }
 
+    private Step processDefinedStep(final TransformFlowStep step) {
+        Step res = catalog.getReadOnlyCatalog().searchStepByName("transform");
```
Comment (igarashitm) [inline kamelet-support/src/main/java/io/kaoto/backend/api/service/step/parser/kamelet/KameletStepParserService.java:303]: This returns null ATM

Comment (Delawen) [reply kamelet-support/src/main/java/io/kaoto/backend/api/service/step/parser/kamelet/KameletStepParserService.java:303]: Right now it loads whatever is on the latest main branch in https://github.com/KaotoIO/camel-component-metadata

Comment (Delawen) [reply kamelet-support/src/main/java/io/kaoto/backend/api/service/step/parser/kamelet/KameletStepParserService.java:303]: There's probably a hundred better ways to do this, but at this point you should add a file like https://github.com/KaotoIO/camel-component-metadata/blob/main/set-body.camel-component.yaml

Comment (Delawen) [reply kamelet-support/src/main/java/io/kaoto/backend/api/service/step/parser/kamelet/KameletStepParserService.java:303]: https://github.com/KaotoIO/camel-component-metadata/blob/main/transform.camel-component.yaml

**Comment by Delawen** [comment]
Ah, it's complaining because duplicated code.


[INFO]     /home/runner/work/kaoto-backend/kaoto-backend/kamelet-support/src/main/java/io/kaoto/backend/KamelPopulator.java line 332
[INFO]     /home/runner/work/kaoto-backend/kaoto-backend/kamelet-support/src/main/java/io/kaoto/backend/KamelPopulator.java line 367

Try to refactor as I did in https://github.com/KaotoIO/kaoto-backend/pull/131/files#diff-50a34f0ddfe536b1002466f6393e5a1421c9c669577071aa27ea7b7be3e09fd7L329-R343

**Comment by Delawen** [comment]
I should separate the EIP in different classes, right now we have conflicts very easily :/

**Comment by igarashitm** [comment]
Thanks for the review! rebased and fixed the duplicate.

**Pull request merged**